psi_cardinality/go/server/server.go

```diff
@@ -201,10 +202,11 @@ func (s *PSICardinalityServer) GetPrivateKeyBytes() (string, error) {
 	rcode := C.psi_cardinality_server_get_private_key_bytes(s.context, &out, &outlen, &err)
 
 	if rcode != 0 {
-		return "", fmt.Errorf("get private keys failed: %v(%v)", s.loadCString(&err), rcode)
+		return nil, fmt.Errorf("get private keys failed: %v(%v)", s.loadCString(&err), rcode)
 	}
 
-	result := s.loadCString(&out)
+    // Convert C array to a Go slice. Private Keys are guaranteed to be 32 bytes long.
+	result := (*[32]byte)(unsafe.Pointer(out))[:outlen:outlen]
```

Comment:
Maybe `result := C.GoBytes(unsafe.Pointer(out), 32)` is easier to read.

Reply:
Nice, I didn't know that function existed. Is there a reference somewhere for all cgo builtins?

Reply:
Fixed.

Reply:
The official docs are at https://golang.org/cmd/cgo/
but is pretty messy, unfortunately.

It's lame, but the best reference is their unit tests :))
https://github.com/golang/go/blob/master/misc/cgo/test/test.go